### ISSUE TITLE
Extend argvals to include vectors and maps

### ISSUE THREAD

**Issue by dcerys** [dollabs/pamela#84]
Currently, values of the arguments to plant functions are defined as:
```
argval   ::= symbol
           | boolean
           | string
           | number
           | safe-keyword
```
Extend this to also include vectors and maps.

**Comment by tmarble** [comment]
@dcerys Is your intention that these collections may be recursive?
In other words is a vector defined as a vector of argvals?
Is a map a set of keyword :: argval mappings (i.e. do we want to restrict the type of keys or not)?
Would there be any reason to not have arbitrarily nested collections?

**Comment by dcerys** [comment]
@tmarble Yes, the ultimate goal is that the values could be collections of arbitrary, heterogeneous values.

**Comment by tmarble** [comment]
@dcerys I have created a sample file for you to evaluate:
https://github.com/tmarble/pamela/blob/issue-84/test/pamela/issue-84.pamela

NOTE: this syntax currently *does not build*. Please advise if this is what you want.

**Comment by dcerys** [comment]
Yes, syntax-wise, this sample file shows examples of the use of vectors and maps (including a vector embedded within a map).

As is often the case, this raises another issue.  See #89.

**Comment by dcerys** [comment]
Since this a clean extension, we'll defer this until we have a pressing project that needs this.
